auth/handler.go

```diff
@@ -276,7 +276,10 @@ func (h *Handler) ServeProviders(w http.ResponseWriter, req *http.Request) {
 	if errutil.HTTPError(req.Context(), w, err) {
 		return
 	}
-	w.Write(data)
+	_, err = w.Write(data)
+	if errutil.HTTPError(req.Context(), w, err) {
+		return
+	}
```

Comment:
This is incorrect; this code will attempt to send and log a 500 status code on a write failure, which occurs when the client has already disconnected, _and_ calling `w.Write` already sends the status code and headers; attempts to do so a second time is also invalid.

Since it's normal/expected to have a write error (e.g., when someone closes a tab), this error should probably be ignored. If there's a way to filter out connection errors, logging would be okay, but it would not be helpful to have errors in the logs every time someone closes a tab.

Reply:
done.  I errored on the side of caution and tried to copy error patterns within the function.  I'm totally fine with ignoring the error in this case if you are.